sdk/resourcemanager/azure-resourcemanager-samples/src/main/java/com/azure/resourcemanager/compute/samples/CloneVirtualMachineToNewRegion.java

```diff
@@ -0,0 +1,314 @@
+ *  - Create an managed virtual machine from PIR image with data disks
+ *  - Create incremental snapshot from the virtual machine's OS and data disks
+ *  - Copy incremental snapshots to the new region
+ *  - Create managed disks from the snapshots of the new region
+ *  - Create virtual machine by attaching the managed disks
+ */
+public final class CloneVirtualMachineToNewRegion {
+    /**
+     * Main function which runs the actual sample.
+     * @param azureResourceManager instance of the azure client
+     * @return true if sample runs successfully
+     */
+    public static boolean runSample(AzureResourceManager azureResourceManager) {
+        final String linuxVMName1 = Utils.randomResourceName(azureResourceManager, "VM1", 15);
+        final String linuxVMName2 = Utils.randomResourceName(azureResourceManager, "VM2", 15);
+        final String managedOSSnapshotName = Utils.randomResourceName(azureResourceManager, "ss-os-", 15);
+        final String managedDataDiskSnapshotPrefix = Utils.randomResourceName(azureResourceManager, "ss-data-", 15);
+        final String managedNewOSDiskName = Utils.randomResourceName(azureResourceManager, "ds-os-nw-", 15);
+        final String managedNewDataDiskNamePrefix = Utils.randomResourceName(azureResourceManager, "ds-data-nw-", 15);
+
+        final String rgName = Utils.randomResourceName(azureResourceManager, "rgCOMV", 15);
+        final String rgNameNew = Utils.randomResourceName(azureResourceManager, "rgCOMV", 15);
+        final String publicIpDnsLabel = Utils.randomResourceName(azureResourceManager, "pip", 15);
+        final String userName = "tirekicker";
+        final String sshPublicKey = Utils.sshPublicKey();
+        final Region region = Region.US_WEST;
+        final Region regionNew = Region.US_EAST;
+
+        final String apacheInstallScript = "https://raw.githubusercontent.com/Azure/azure-sdk-for-java/main/sdk/resourcemanager/azure-resourcemanager-samples/src/main/resources/install_apache.sh";
+        final String apacheInstallCommand = "bash install_apache.sh";
+        List<String> apacheInstallScriptUris = new ArrayList<>();
+        apacheInstallScriptUris.add(apacheInstallScript);
+
+        try {
+            //=============================================================
+            // Create a Linux VM using a PIR image with managed OS and data disks and customize virtual
+            // machine using custom script extension
+
+            System.out.println("Creating a un-managed Linux VM");
+
+            VirtualMachine linuxVM = azureResourceManager.virtualMachines().define(linuxVMName1)
+                .withRegion(region)
+                .withNewResourceGroup(rgName)
+                .withNewPrimaryNetwork("10.0.0.0/28")
+                .withPrimaryPrivateIPAddressDynamic()
+                .withNewPrimaryPublicIPAddress(publicIpDnsLabel)
+                .withPopularLinuxImage(KnownLinuxVirtualMachineImage.UBUNTU_SERVER_16_04_LTS)
```

Comment:
If possible use a later version.

Reply:
Switched to use 18_04. I saw latest ubuntu version in portal is 20.04, should we added it to `KnownLinuxVirtualMachineImage`?

Reply:
Yeah, if it can be listed, we should add it. Anyway that would be another PR.